fmp_server.py

```diff
@@ -26,9 +26,61 @@
 from other.utils import hexdump, server_base, server_main
 
 
+def construct_broadcast(text="Hello, World!"):
+    assert len(text) > 0
+    MAX_CHAR_LENGTH = 30
+    broadcast_segments = []
+    words = text.split(" ")
+
+    mod_words = []
+
+    for word in words:
+        curr_word = word
+        while len(curr_word) > 30:
+            mod_words.append(curr_word[0:30])
+            curr_word = curr_word[30:]
+        if len(curr_word) > 0:
+            mod_words.append(curr_word)
+
+    curr_broadcast = mod_words[0]
+    for word in mod_words[1:]:
+        if len(curr_broadcast) + 1 + len(word) <= 30:
+            curr_broadcast += " "
+            curr_broadcast += word
+        else:
+            broadcast_segments.append(curr_broadcast)
+            curr_broadcast = word
+    if len(curr_broadcast) > 0:
+        broadcast_segments.append(curr_broadcast)
```

Comment:
- Isn't there a simpler way to perform this? It looks like a lot of code for a simple task.
- Didn't Capcom used newlines instead of creating a message for each line?

Reply:
It has to trim down lengthy words to segments that are at most the maximum character length, then do the math word-by-word to see if each one fits onto the current line when including a space. Those are 2 distinct processes, thus the 2 different for loops.

I DID completely forget to use the constant `MAX_CHAR_LENGTH` I defined though. So, oops.

Reply:
Also, I don't think Capcom did? The messages always showed up as distinct lines, each with their own sender name.

Reply:
> Also, I don't think Capcom did? The messages always showed up as distinct lines, each with their own sender name.

That seems correct, they are sending 4 messages as part of their announcement as seen on these videos:
 - https://youtu.be/ZTwdF9e9Kk0?t=2943
 - https://youtu.be/K5dX4KhUFUM?t=1570

![image](https://user-images.githubusercontent.com/7890055/201517318-451ce90b-bbb3-4e75-87ba-ae100f697992.png)
![image](https://user-images.githubusercontent.com/7890055/201517338-33c680e7-2796-4108-8d88-c9a0b9844a5a.png)